pkg/ccl/crosscluster/logical/create_logical_replication_stmt.go

```diff
@@ -22,17 +37,156 @@ func init() {
+
+		if !stmt.Options.IsDefault() {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication stream options are not yet supported")
+		}
+		if stmt.On.Database != "" {
+			return errors.UnimplementedErrorf(issuelink.IssueLink{}, "logical replication streams on databases are unsupported")
+		}
+		if len(stmt.On.Tables) != len(stmt.Into.Tables) {
+			return pgerror.New(pgcode.InvalidParameterValue, "the same number of source and destination tables must be specified")
+		}
+
+		streamAddress := crosscluster.StreamAddress(from)
+		streamURL, err := streamAddress.URL()
+		if err != nil {
+			return err
+		}
+		streamAddress = crosscluster.StreamAddress(streamURL.String())
+
+		client, err := streamclient.NewStreamClient(ctx, streamAddress, p.ExecCfg().InternalDB)
+		if err != nil {
+			return err
+		}
+		defer func() {
+			_ = client.Close(ctx)
+		}()
+
+		if err := client.Dial(ctx); err != nil {
+			return err
+		}
+
+		var targetsDescription string
+		srcTableNames := make([]string, len(stmt.On.Tables))
+
+		repPairs := make([]jobspb.LogicalReplicationDetails_ReplicationPair, len(stmt.Into.Tables))
+		for i := range stmt.On.Tables {
+
+			dstObjName, err := stmt.Into.Tables[i].ToUnresolvedObjectName(tree.NoAnnotation)
+			if err != nil {
+				return err
+			}
+			dstTableName := dstObjName.ToTableName()
+			prefix, td, err := resolver.ResolveMutableExistingTableObject(ctx, p, &dstTableName, true, tree.ResolveRequireTableDesc)
+			if err != nil {
+				return err
+			}
+
+			repPairs[i].DstDescriptorID = int32(td.GetID())
+
+			tbNameWithSchema := tree.MakeTableNameWithSchema(
+				tree.Name(prefix.Database.GetName()),
+				tree.Name(prefix.Schema.GetName()),
+				tree.Name(td.GetName()),
+			)
+
+			srcTableNames[i] = stmt.On.Tables[i].String()
+
+			if i == 0 {
+				targetsDescription = tbNameWithSchema.FQString()
+			} else {
+				targetsDescription += ", " + tbNameWithSchema.FQString()
+			}
+		}
+
+		spec, err := client.CreateForTables(ctx, &streampb.ReplicationProducerRequest{
+			TableNames: srcTableNames,
+		})
+		if err != nil {
+			return err
+		}
+
+		for i, name := range srcTableNames {
+			repPairs[i].SrcDescriptorID = int32(spec.TableDescriptors[name].ID)
+		}
+
+		jr := jobs.Record{
+			JobID:       p.ExecCfg().JobRegistry.MakeJobID(),
+			Description: fmt.Sprintf("LOGICAL REPLICATION STREAM into %s", targetsDescription),
+			Username:    p.User(),
+			Details: jobspb.LogicalReplicationDetails{
+				StreamID:             uint64(spec.StreamID),
+				SourceClusterID:      spec.SourceClusterID,
+				ReplicationStartTime: spec.ReplicationStartTime,
+				TargetClusterConnStr: string(streamAddress),
+				ReplicationPairs:     repPairs,
+				TableNames:           srcTableNames},
+			Progress: jobspb.LogicalReplicationProgress{},
+		}
+
+		if _, err := p.ExecCfg().JobRegistry.CreateAdoptableJobWithTxn(ctx, jr, jr.JobID, p.InternalSQLTxn()); err != nil {
+			return err
+		}
+		p.ExecCfg().JobRegistry.NotifyToAdoptJobs()
```

Comment:
I know this was in the existing code, but I think this was an artifact of me not thinking things through. The job is created only when the txn commits so I don't think this does much for us.